### PR TITLE
Android improvements

### PR DESCRIPTION
**Pull request by Drabble**
This PR includes multiples improvements for Android:

1. Add x86 as a target in addition to x86_64 and arm64.
2. Unify minSDK to 21 which is the lowest version that support OpenGL ES 3.1 on Android.
3. Unify targetSDK to 31.
4. Update [https://github.com/mozilla/rust-android-gradle](https://github.com/mozilla/rust-android-gradle) to version 0.9.3 and remove the custom config related to [this issue](https://github.com/mozilla/rust-android-gradle/issues/91) since it has been fixed in the new version. This custom configuration didn't work on non-linux operating systems.
5. Remove limits and leave only `downlevel_defaults` for Android because `downlevel_defaults` has the defaults for OpenGL ES 3.1.

## 💻 Examples

\-

## 🚨 Test instructions

Run the Android demo project with Android Studio. The documentation has been improved in #102.

## ✔️ PR Todo

- [x] Make all the improvements in the list above.

### CODE REVIEW

**Comment by Drabble** [comment]
There is an error in the github action for Android. Here is the relevant part of the log:

> Task :lib:cargoBuildX86
error[E0463]: can't find crate for `core`
  |
  = note: the `i686-linux-android` target may not be installed
  = help: consider downloading the target with `rustup target add i686-linux-android`
  = help: consider building the standard library from source with `cargo build -Zbuild-std`

The command `./gradlew assembleDebug` in `just build-android` probably builds for all the targets. The new target x86 doesn't have access to the right toolchain. I will try to fix it tomorrow, and I'll add check for x86 in the github action.

**Comment by maxammann** [comment]
The toolchain needs to be added here: https://github.com/maplibre/maplibre-rs/blob/8b5a4d241fd8e8e5690e0804ec8aa2268d59f069/justfile#L35

File: android/gradle/.android/advancedFeatures.ini [hunk 1]
```diff
@@ -0,0 +1 @@
+GLESDynamicVersion = on
```
Comment (maxammann) [inline android/gradle/.android/advancedFeatures.ini:1]: How is that file used? I think this is for the emulator, right? how does my local installation find this?

**Review by maxammann — APPROVED** [review]
Looks good! thanks!

File: android/gradle/.android/advancedFeatures.ini (hunk 1 above)
Comment (Drabble) [reply android/gradle/.android/advancedFeatures.ini:1]: This file is needed for the emulator to enable OpenGL ES 3.1 I believe.
I found the solution here: https://stackoverflow.com/a/44552548/2794211

**Comment by Drabble** [comment]
> The toolchain needs to be added here:
> 
> https://github.com/maplibre/maplibre-rs/blob/8b5a4d241fd8e8e5690e0804ec8aa2268d59f069/justfile#L35

I have updated the justfile to install the missing toolchain. Why are we installing it from nightly though?

File: android/gradle/.android/advancedFeatures.ini (hunk 1 above)
Comment (maxammann) [reply android/gradle/.android/advancedFeatures.ini:1]: But this file should be in `~/.android/`. That means users need to install this manually/make the changes manually. So we need to document this but not add this file.

**Comment by maxammann** [comment]
> I have updated the justfile to install the missing toolchain. Why are we installing it from nightly though?

Because we use a nightly feature which rebuilds the stdlib.

**Review by maxammann — CHANGES_REQUESTED** [review]
.

File: android/gradle/.android/advancedFeatures.ini (hunk 1 above)
Comment (Drabble) [reply android/gradle/.android/advancedFeatures.ini:1]: Oh that's right, I had read it wrong! My problem solved itself when I added the file, but if the folder it should live in is ~/.android, it is probably something else that solved my problem... I will try again to see if it works without the file and I guess we can remove it if it doesn't help. I'll see if there is another special configuration I made for the emulator in Android Studio and document it. I'll do that next week probably.

Comment (Drabble) [reply android/gradle/.android/advancedFeatures.ini]: I have removed the file as it was not used anyway.

**Review by maxammann — APPROVED** [review]
Looks good!

**Pull request merged**